### ISSUE TITLE
multiple asyncPropsExtenders defined on same path not working, only first route works

### ISSUE THREAD

**Issue by giuliaghisini** [plone/volto#3555]
if i have defined multiple asynPropExtenders on same route, for example 

```
config.settings.asyncPropsExtenders = [
    ...(config.settings.asyncPropsExtenders ?? []),
   {
      **path: '/',**
      extend: (dispatchActions) => {
        if (
          dispatchActions.filter((asyncAction) => asyncAction.key === '**dropdown-menu**').length === 0
        ) {
          dispatchActions.push({
            key: 'dropdownmenu',
            promise: ({ location, store: { dispatch } }) => __SERVER__ && dispatch( getDropdownMenu()),
          });
        }
        return dispatchActions;
      },
    },
    {
      **path: '/',**
      extend: (dispatchActions) => {
        if (
          dispatchActions.filter((asyncAction) => asyncAction.key === '**subsite**').length === 0
        ) {
          dispatchActions.push({
            key: 'subsite',
            promise: ({ location, store: { dispatch } }) => __SERVER__ && dispatch( getSubsite()),
          });
        }
        return dispatchActions;
      },
    },
  ];
```

only first route is added to extenders 
because matchRoutes(asyncPropsExtenders, pathname); returns only the first route... 
This is not correct, because other asynPropExtenders doesn't works.. 

So, in https://github.com/plone/volto/blob/72c0a8f2c50b00390e591dd53e3c8c97aa02e0fb/src/helpers/AsyncConnect/index.js#L60

i suggest this change:
```
  //const extenders = matchRoutes(asyncPropsExtenders, pathname);
  let extenders = [];
  asyncPropsExtenders.forEach((ext) => {
    const match = matchRoutes([ext], pathname);
    extenders = [...extenders, ...match];
  });
```

this make things working...

@tiberiuichim @sneridagh what do you think?

**Comment by tiberiuichim** [comment]
I'm ok with your proposal @giuliaghisini

**Comment by pnicolli** [comment]
@tiberiuichim @giuliaghisini was this fixed by #3645 ?
